src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceGeneratedFileInfo.cs

```diff
@@ -25,20 +25,16 @@ internal sealed class MetadataAsSourceGeneratedFileInfo
 
         private readonly ParseOptions? _parseOptions;
 
-        public MetadataAsSourceGeneratedFileInfo(string rootPath, Project sourceProject, INamedTypeSymbol topLevelNamedType, bool signaturesOnly)
+        public MetadataAsSourceGeneratedFileInfo(string rootPath, Workspace sourceWorkspace, Project sourceProject, INamedTypeSymbol topLevelNamedType, bool signaturesOnly)
```

Comment:
the core of the change was to pass `Workspace sourceWorkspace` into here.  The rest of the PR is just flowing that outwards so it can be passed in.

Reply:
What is it actually used for though? Is it important?

Reply:
it's definitely important.  that said, will have meeting tomorrow to see if we can just get rid of MASWorkspace entirely :)

Reply:
I meant the source workspace. Is it just for getting services?

Reply:
Looks like maybe the workspace could be a parameter on the SymbolMappingService methods, and source workspace might not be needed here?

Reply:
> I meant the source workspace. Is it just for getting services?

It's for two purposes. One is for the services. The other is to know the original workspace the user started from so operations (like find refs) can pop back out to that)

Reply:
> Looks like maybe the workspace could be a parameter on the SymbolMappingService methods, and source workspace might not be needed here?

That was actually the exact original thing I was trying. However the callers do not have this source workspace.  Instead, they are starting from a doc/workspace for the *MAS* workspace. The reason we store this here is so that the *caller* can pop back to the actual real host workspace. 

Note: tomorrow's meeting will be about potentially removing all of this :-)